src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -399,6 +399,61 @@ QDateTime QgsWmsSettings::parseWmstDateTimes( const QString &item )
     return QDateTime::fromString( item, Qt::ISODate );
 }
 
+QgsDateTimeRange QgsWmsSettings::parseWmtsTimeValue( const QString &value, QgsWmtsTileLayer::WmtsTimeFormat &format )
```

Comment:
Ah, this is a little different because the incoming strings are just single time values, not the standard "start/end/interval" format. We return a range from this function because we do some magic tricks like interpreting the string "2020" as meaning "ALL of 2020" (and not just 2020-01-01), or interpreting "20200304" as "ALL OF 2020-03-04" ie "2020-03-04 0:00:00 to 2020-03-04 23:59:59"). I think it's sufficiently ugly logic that we'd want to keep it specific to interpreting WMTS date values and not pass it off as a generic "any string to date" conversion method.